### PR TITLE
Fix perpendicular caluclations of Vector2f and Vector2d

### PR DESCRIPTION
**Pull request by ieperen3039**
the calculation for y used the new value of x

### CODE REVIEW

File: src/org/joml/Vector2d.java [hunk 1]
```diff
@@ -497,8 +497,9 @@ public Vector2dc getToAddress(long address) {
      * @return this
      */
     public Vector2d perpendicular() {
-        this.x = y;
+        float xTemp = y;
```
Comment (ieperen3039) [inline src/org/joml/Vector2d.java:500]:
```suggestion
        double xTemp = y;
```

**Comment by httpdigest** [comment]
Thanks for finding and fixing! Those kind of bugs were all introduced by https://github.com/JOML-CI/JOML/commit/50359e911f4ac6f8a163764708aaa25c2c89a3fc#diff-5f25b65b863f5962b1afa652e0479798L507 (see the Vector2d diff).
Sadly, this was a simple "inline refactor" done via Eclipse, which I assumed would be correct. However it totally screwed things up in this case.

**Comment by httpdigest** [comment]
Yeah, Eclipse's method invocation inlining refactor (Shift+Alt+I) is to blame. Take for example this simplified case:
```java
public class Vector2f {
  public float x, y;
  public Vector2f perpendicular() {
    return this.set(y, -x);
  }
  private Vector2f set(float x, float y) {
    this.x = x;
    this.y = y;
    return this;
  }
}
```
and inline the `this.set(y, -x)` invocation in `perpendiculat()`. The result will be:
```java
public Vector2f perpendicular() {
  this.x = y;
  this.y = -x;
  return this;
}
```
which is totally wrong here.

**Comment by ieperen3039** [comment]
I believe IntelliJ does it correctly

**Pull request merged**